citesphere/src/main/java/edu/asu/diging/citesphere/web/user/authorities/AuthorityEntryController.java

```diff
@@ -71,8 +84,71 @@
             logger.info("Not a valid URI: " + uri, e);
             return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
         }
-     
-        entry = authorityService.create(entry, (IUser)authentication.getPrincipal());
+
+        entry = authorityService.create(entry, (IUser) authentication.getPrincipal());
+        return new ResponseEntity<IAuthorityEntry>(entry, HttpStatus.OK);
+    }
+    
+    @RequestMapping(value = "/auth/authority/conceptpower/create", method = RequestMethod.POST)
+    public ResponseEntity<IAuthorityEntry> createConceptpowerAuthorityEntry(Authentication authentication,
+            @RequestParam("uri") String uri) {
+        IAuthorityEntry entry = null;
+        for(IAuthorityEntry authorityEntry : authorityConceptpowerResult.getFoundAuthorities()) {
+            if(authorityEntry.getUri().equals(uri)) {
+                entry = authorityService.create(authorityEntry, (IUser) authentication.getPrincipal());
+                break;
+            }
+        } 
```

Comment:
I would use the stream api and filter the list, then use the result to create a new authority.